### PR TITLE
[SLDEV-13926] - starting an otel span on each test + refactoring + bugfix + selenium support

### PR DESCRIPTION
**Pull request by shaicantor**
* refactoring
  * base_url is extracted from token and not explicitly set
  * extracted logic to methods for clearer code
* fixed a huge bug in TIA. Tests were not actually skipped, just marked as skipped.
* each test start now opens an otel span. added test_something.robot as an example test

### CODE REVIEW

File: robot-custom-integration/sl_python_robot/SLListener.py [hunk 1]
```diff
@@ -93,9 +121,21 @@ def end_test_session(self):
...
-   
+
+    def start_span(self, test_name):
+        test_span = self.spans.get(test_name)
+        if test_span:
+            return test_span
+        span = tracer.start_span(test_name)
+        ctx = trace.set_span_in_context(span, context.get_current())
+        ctx = baggage.set_baggage("x-sl-test-name", test_name, ctx)
+        ctx = baggage.set_baggage("x-sl-execution-id", self.test_session_id, ctx)
```
Comment (alonweiss-sl) [inline robot-custom-integration/sl_python_robot/SLListener.py]: please change to `x-sl-test-session-id`

**Comment by marcam-sl** [comment]
@alonweiss-sl why is this PR in the Public Repo of TIA API integration samples?

File: robot-custom-integration/sl_python_robot/SLListener.py (hunk 1 above)
Comment (shaicantor) [reply robot-custom-integration/sl_python_robot/SLListener.py]: done

File: robot-custom-integration/sl_python_robot/SLListener.py [hunk 2]
```diff
@@ -35,11 +55,27 @@ def end_suite(self, date, result):
         self.send_test_results(test_results)
         self.end_test_session()
 
-    #--- Sealights API helpers ---
-    
+    def start_test(self, data, result):
+        test_name = data.name
```
Comment (kristijanstefanoski) [inline robot-custom-integration/sl_python_robot/SLListener.py]: We need to make sure `test_name` is URL encoded in case of white spaces etc..

Comment (kristijanstefanoski) [reply robot-custom-integration/sl_python_robot/SLListener.py]: I will take care of this in the OTEL plugin in the browser-agent, will encode always the testName if not already

Comment (shaicantor) [reply robot-custom-integration/sl_python_robot/SLListener.py]: @kristijanstefanoski , done here.
https://sealights.atlassian.net/browse/SLDEV-14387

**Review by alonweiss-sl — APPROVED** [review]

**Pull request merged**